dom/src/main/scala/fs2/dom/Window.scala

```diff
@@ -65,6 +72,15 @@ object Window {
 
       def document: HtmlDocument[F] = window.document.asInstanceOf[HtmlDocument[F]]
 
+      def requestAnimationFrame: F[FiniteDuration] = F.async[FiniteDuration] { cb =>
```

Comment:
I'm a bit unsure about the naming. What we are exposing here seems much less powerful than the JS version. Indeed, while the JS version allows us to execute arbitrary code before the next repaint, this version only guarantees that the effect completes *after* the first callback queued by `requestAnimationFrame` fires. However, `requestAnimationFrame >> someEffect` clearly doesn't guarantee that `someEffect` runs before the next repaint. Consequently, this version seems useful only as a kind of "metronome" for driving a stream at a rate not exceeding the display refresh rate (which can be very useful, of course). And if that's the case, then perhaps the stream shouldn't be called `animationFrames` (since it doesn't emit animation frames), but rather something like `awakeEveryAnimationFrame` in keeping with fs2 naming conventions.

Reply:
Ah right, this is the microtask vs macrotask issue again - I take everything back 😅. One last comment, if you do decide to name it `awakeEveryAnimationFrame`, then to be consistent with fs2 it should probably emit the duration since the beginning of the stream:
```scala
    Stream.eval(t.monotonic).flatMap { start =>
      animationFrames >> Stream.eval(
        t.monotonic.map(_ - start)
      )
    }
```